msgpack-core/src/main/java/org/msgpack/core/MessagePackFactory.java

```diff
@@ -0,0 +1,203 @@
+//
+// MessagePack for Java
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+package org.msgpack.core;
+
+import org.msgpack.core.buffer.ArrayBufferInput;
+import org.msgpack.core.buffer.ChannelBufferInput;
+import org.msgpack.core.buffer.ChannelBufferOutput;
+import org.msgpack.core.buffer.InputStreamBufferInput;
+import org.msgpack.core.buffer.OutputStreamBufferOutput;
+import org.msgpack.core.buffer.MessageBufferInput;
+import org.msgpack.core.buffer.MessageBufferOutput;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.channels.WritableByteChannel;
+import java.nio.channels.ReadableByteChannel;
+
+import java.nio.charset.CodingErrorAction;
+import static org.msgpack.core.Preconditions.checkArgument;
+
+public class MessagePackFactory
```

Comment:
Why don't you divide this class into `MessagePackerFactory` and `MessageUnpackerFactory` so that we can remove verbose prefixes such as `packerXXXXX` and `unpackerXXXXX`?

Also, I found msgpack-java 0.7 users already use jackson-dataformat-msgpack and they've written this code `new ObjectMapper(new MessagePackFactory())` in their source code. If we separate this class into `MessagePackerFactory` and `MessageUnpackerFactory`, we don't need to rename `org.msgpack.jackson.dataformat.MessagePackFactory` to `org.msgpack.jackson.dataformat.MessagePackFormatFactory` and the users don't need to change their source code.